reference/promotion_settings.v3.yml

```diff
@@ -0,0 +1,299 @@
+      summary: Get Global Promotion Settings
+      description: Returns global promotion settings.
+      operationId: getGlobalPromotionSettings
+      tags:
+        - Promotion Settings
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                allOf:
+                  - $ref: "#/components/schemas/GlobalPromotionSettings"
+              examples:
+                EnableZeroPriceProductsToTriggerPromotions:
+                  $ref: "#/components/examples/EnableZeroPriceProductsToTriggerPromotions"
+        "400":
+          description: Bad request. Authentication Required.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse400"
+    put:
+      summary: Update Global Promotion Settings
+      description: Updates global promotion settings.
+      operationId: updateGlobalPromotionSettings
+      parameters:
+        - $ref: "#/components/parameters/ContentType"
+      tags:
+        - Promotion Settings
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/GlobalPromotionSettings"
+            examples:
+              EnableZeroPriceProductsToTriggerPromotions:
+                $ref: "#/components/examples/EnableZeroPriceProductsToTriggerPromotions"
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                allOf:
+                  - $ref: "#/components/schemas/GlobalPromotionSettings"
+        "400":
+          description: Bad request. Authentication Required.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse400"
+        "403":
+          description: Forbidden. Operation cannot be performed with existing privileges.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse403"
+        "422":
+          description: "Promotion settings data is not valid. This is the result of missing required fields, or of invalid data. See the response for more details."
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse422"
+
+components:
+  schemas:
+    GlobalPromotionSettings:
+      type: object
+      properties:
+        promotions_apply_on_products_with_custom_product_price:
+          type: boolean
+          example: false
+          default: false
+          description: |-
+            - When the property is set to "true", promotions will apply for products with custom pricing.
+            - When the property is set to "false", promotions will not apply for products with custom pricing.
+            - Note: This field can only be edited for Enterprise Accounts
```

Comment:
Given we have a few examples of fields specific to enterprise accounts, it's probably worthwhile defining what the behaviour is for non-enterprise accounts somewhere in this schema. I believe we brought this up when fleshing out some related tickets in a refinement session.

Some options: 
- Do we throw an error if a non-enterprise merchant tries to pass a non-default setting to an enterprise-only field?
- Do we not persist the change and just ignore it?
- Do we ignore it but offer some kind of feedback in the payload about them attempting to edit an enterprise-only field?

Reply:
@bc-vincent-zhao did we happen to officially decide on an approach?

Reply:
From memory we decided to throw an error if the value in the PUT request is different to the current value for non enterprise plans.

Reply:
what @BC-SEven said ^ seems sensible.